### ISSUE TITLE
setState on unmount component

### ISSUE THREAD

**Issue by midori0507** [ReactTraining/react-idle#12]
when i do a history push outside of current component with react-router, the counter still counting
and this appear

```
Warning: setState(...): Can only update a mounted or mounting component. This usually means you called setState() on an unmounted component. This is a no-op. Please check the code for the Idle component.
```

**Comment by midori0507** [comment]
Related https://github.com/ReactTraining/react-idle/pull/9 ?

**Comment by newyork-anthonyng** [comment]
@midori0507 yes, #9 should fix this warning. Let’s leave this issue open for others who run into the same issue; until it gets merged.

**Comment by irandamay** [comment]
Also running into this issue, but I don't think the #9 fix fixes it.

I also have a history push in my timers.  There is one on a login page that after 30s sends it back to a welcome/landing page so that the app is not just sitting there at a login page all the time if someone leaves it like that, and there is one when the user is logged in that logs them out after 5 minutes.  

When my user logs in and gets redirected from the login page to the user's home page, even with the #9 fix, the timer from the login page keeps running and sends them back to the landing page every 30s, with this warning in the console every time.  Since the redirect from the login page doesn't log them out, they can navigate back to the user pages but still the timer continues to run and send them back to the landing every 30s.

**Comment by newyork-anthonyng** [comment]
@irandamay Do you have an example project that reproduces this issue?

**Comment by irandamay** [comment]
@newyork-anthonyng Oops, when I reported this initially, I didn't have the time to extract out the bare minimum of what I was doing and then I completely forgot about it.

I am not sure what I was seeing before, but I believe that #9 does indeed fix the warning.  My timers are not stopping, but I guess that's a different issue...
